### DISCUSSION TITLE
Multi temporal raster data (netCDF) in dfs2

### DISCUSSION THREAD

**Discussion by adrienmichez** [DHI/mikeio#407] [Q&A]
Hi,

It would be nice to have an example script to import multi temporal raster data (usually stored in netCDF files) in dfs2 with the python package mikeio.

A lot of users are using satellite imagery as input of mikeshe projects.

And the [DHI netCDF tool](https://github.com/DHI/netcdf-client) is no longer updated

**Comment by adrienmichez** [comment]
[A typical netCDF file](https://filesender.belnet.be/?s=download&token=8d602e1d-1ec0-4cb5-97c0-2829e928b1f8)

**Reply by ecomodeller** [reply]
One of the problems with automatic conversion of any temporal raster data is the time axis, which in this case is monthly.
![image](https://user-images.githubusercontent.com/614215/187134300-fdf2ae42-eae6-4570-ab75-f2f714226e24.png)

In order to create an equidistant time axis with the same number of seconds, the dates will not match the original file, which in this case is probably perfectly fine, but I prefer to leave these decisions to the user of MIKE IO, rather than trying to come up with workarounds deep inside some high-level function.

I think the [Bathymetry notebook](https://github.com/DHI/mikeio/blob/main/notebooks/Dfs2%20-%20Bathymetry.ipynb) should get you pretty far. 

📢 Anyone working with multi-temporal raster data is of course free to contribute with an example notebook showing how they have handled the conversion.

**Reply by adrienmichez** [reply]
Ok, I'll have a look to the notebook ASAP.

Just for my information, is equidistant time axis mandatory to work with mikeSHE?

**Reply by ecomodeller** [reply]
Maybe @JesperGr knows?

**Comment by jsmariegaard** [comment]
Thanks. I agree. Notice also this PR: https://github.com/DHI/mikeio/pull/369

**Comment by watermain** [comment]
MShe is able to read and write non-equidistant time axes for dfs0, dfs2 and dfs3 files. Read values will be linearly interpolated between times. As it is not supported to create these files with the MZ grid editor though it is not being used very much, so before depending on this for a proposal/project I´d recommend to do your own testing. ~uha

**Comment by adrienmichez** [comment]
I did some testing. What do you think? Is it OK? 
I'm not sure about the re indexing of the Y dimension.  Feel free to edit or use it for a notebook. @ecomodeller I never did a notebook but I'm OK to try with some help
To proper format the file, we should name the parameter (here Leaf Area Index values from satellite optical imagery). I don't know how to access to the item list 
```
import xarray
import mikeio

# loading the file
# I don't how to do it but online version on my github:  https://github.com/adrienmichez/PUBLIC/raw/main/LAI_300m_Vesdre_10days_copernicus_MONTHLY_AGG_2021.nc

ds = xarray.open_dataset("C:\DATA_TMP\GIS\VESDRE\LAI_300m_Vesdre_10days_copernicus_MONTHLY_AGG_2021.nc")

# because ascending values are mandatory for mikeio.Grid2D()
ds = ds.reindex(y = ds.y[::-1])

#creating the geometry
geometry = mikeio.Grid2D(x=ds.x.values, y = ds.y.values, projection="LONG/LAT")

#creating a mikeio data array
da = mikeio.DataArray( data = ds.LAI.values, geometry = geometry, 
                      time=ds.t.values,
                      dims=("time", "y","x"))

# save it
da.to_dfs('C:/TMP/test.dfs2')
ds2 = mikeio.read("C:/TMP/test.dfs2") #looks OK to me

```

**Reply by ecomodeller** [reply]
Can you cite the source of the example data?

**Reply by adrienmichez** [reply]
Yes: https://land.copernicus.eu/global/products/lai

**Reply by adrienmichez** [reply]
And the aggregation was performed with [openEO](https://openeo.org/)

**Reply by adrienmichez** [reply]
@ecomodeller did you get time to look at my snippet? Sounds correct to you? Tell me if you need a hand for the jupyter notebook!

**Reply by ecomodeller** [reply]
Seems correct.
